src/physics/cam/nn_convection_flux.F90

```diff
@@ -0,0 +1,525 @@
+            !! Surface precipitation due to autoconversion and sedimentation
+
+        ! -----------------------------------
+        ! Local Variables
+        ! -----------------------------------
+        integer  i, k, dim_counter, out_dim_counter
+        integer nx
+            !! Number of x points in a subdomain
+        integer nzm
+            !! Number of z points in a subdomain - 1
+        ! real(8) :: omn
+        !     !! variable to store result of omegan function
+        !     !! Note that if using you will need to import omegan() from SAM_consts_mod
+
+        ! Other variables
+        real(8),   dimension(nrf) :: omp, fac
+        real(8),   dimension(size(tabs_i, 2)) :: qsat, irhoadz, irhoadzdz
+
+        ! -----------------------------------
+        ! variables for NN
+        ! -----------------------------------
+        real(4), dimension(n_inputs) :: features
+            !! Vector of input features for the NN
+        real(4), dimension(n_outputs) :: outputs
+            !! vector of output features from the NN
+        ! NN outputs
+        real(8),   dimension(nrf) :: t_flux_adv, q_flux_adv, q_tend_auto, &
+                                  q_sed_flux
+        ! Output variable t_rad_rest_tend is also an output from the NN (defined above)
+
+        nx = size(tabs_i, 1)
+        nzm = size(tabs_i, 2)
+
+        ! Check that we have initialised all of the variables.
+        if (do_init) call error_mesg('NN has not yet been initialised using nn_convection_flux_init.')
+
+        ! Define useful variables relating to grid spacing to convert fluxes to tendencies
+        do k=1,nzm
+            irhoadz(k) = dtn/(rho(k)*adz(k)) !  Temporary factor for below
+            irhoadzdz(k) = irhoadz(k)/dz ! 2.0 * dtn / (rho(k)*(z(k+1) - z(k-1))) [(kg.m/s)^-1]
+        end do
+
+        ! The NN operates on atmospheric columns which have been flattened into 2D
+        do i=1,nx
+            ! Initialize variables
+            features = 0.
+            dim_counter = 0
+            outputs = 0.
+            t_rad_rest_tend(i,:) = 0.
+            t_flux_adv = 0.
+            q_flux_adv = 0.
+            t_delta_adv(i,:) = 0.
+            q_delta_adv(i,:) = 0.
+            q_tend_auto = 0.
+            t_delta_auto(i,:) = 0.
+            q_delta_auto(i,:) = 0.
+            q_sed_flux = 0.
+            t_delta_sed(i,:) = 0.
+            q_delta_sed(i,:) = 0.
+            precsfc(i) = 0.
+            omp = 0.
+            fac = 0.
+
+            !-----------------------------------------------------
+            ! Combine all features into one vector
+
+            ! Add temperature as input feature
+            features(dim_counter+1:dim_counter + input_ver_dim) = real(tabs_i(i,1:input_ver_dim),4)
+            dim_counter = dim_counter + input_ver_dim
+
+            ! Add non-precipitating water mixing ratio as input feature using random forest (rf) approach from earlier paper
```

Comment:
Is the commented-out part below still needed or can it be removed?

Reply:
As above it is unused at present, but there is a possibility it may be reimplemented, so I would encourage leaving in for now and opening an issue to track and clean in future.